sdc/io/csv_ext.py

```diff
@@ -398,6 +399,18 @@ def to_varname(string):
     return re.sub(r'\W|^(?=\d)','_', string)
 
 
+def pyarrow_cpu_count_equal_numba_num_treads(func):
+    """Decorator. Set pyarrow cpu_count the same as NUMBA_NUM_THREADS."""
+    def wrapper(*args, **kwargs):
+        cpu_count = pyarrow.cpu_count()
+        pyarrow.set_cpu_count(numba.config.NUMBA_NUM_THREADS)
+        result = func(*args, **kwargs)
```

Comment:
nitpick: if wrapped func raise exception you will miss second call to `set_cpu_count` (but I am not sure numba supports finally)

Reply:
@Hardcode84 as far as I get it this code is not jitted anyway, so `finally` should work